packages/astro/src/vite-plugin-astro-server/base.ts

```diff
@@ -52,11 +53,15 @@ export function baseMiddleware(
 		const publicPath = new URL('.' + req.url, config.publicDir);
 		fs.stat(publicPath, (_err, stats) => {
 			if (stats) {
-				const expectedLocation = new URL('.' + url, devRootURL).pathname;
+				const publicDir = appendForwardSlash(
+					removeTrailingForwardSlash(config.publicDir.pathname).split('/').pop()!
+				);
```

Comment:
If `publicDir` is `some/static/path`, wouldn't this only show the last part `path`? I think we could try to construct a relative path using `path.relative(fileUrlToPath(config.publicDir), fileUrlToPath(config.root))` perhaps so we can get the other directories. `URL` doesn't seem to make this easy.

Reply:
That seems like a great idea, I'll try tomorrow, I should have time then👍

Reply:
Done, Great idea! : )